client/adventureMap/AdventureMapInterface.cpp

```diff
@@ -153,15 +153,30 @@ void AdventureMapInterface::deactivate()
 void AdventureMapInterface::showAll(Canvas & to)
 {
 	CIntObject::showAll(to);
+	dim(to);
 	LOCPLINT->cingconsole->show(to);
 }
 
 void AdventureMapInterface::show(Canvas & to)
 {
 	CIntObject::show(to);
+	dim(to);
 	LOCPLINT->cingconsole->show(to);
 }
 
+void AdventureMapInterface::dim(Canvas & to)
+{
+	for (auto window : GH.windows().findWindows<IShowActivatable>())
+	{
+		std::shared_ptr<AdventureMapInterface> casted = std::dynamic_pointer_cast<AdventureMapInterface>(window);
+		if (!casted && !window->isPopupWindow())
+		{
+			to.drawColor(Rect(0, 0, GH.screenDimensions().x, GH.screenDimensions().y), ColorRGBA(0, 0, 0, std::clamp<int>(settings["adventure"]["backgroundDimLevel"].Integer(), 0, 255)));

```

Comment:
Is it possible to precompute color in constructor to avoid repeating during rendering?

Reply:
also, rect could be moved out of cycle

Reply:
Sorry, what?
Both color and rect are generated only once - during single call to draw

Reply:
```
if (!casted && !window->isPopupWindow())
 {
 	to.drawColor(Rect(0, 0, GH.screenDimensions().x, GH.screenDimensions().y), ColorRGBA(0, 0, 0, std::clamp<int>(settings["adventure"]["backgroundDimLevel"].Integer(), 0, 255)));
 	logGlobal("this line is called");
```

```
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mError: Failed to find file zvs/Lib1.res/NonLiving
[0m[1;31mError: failed to load image zvs/Lib1.res/NonLiving
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mFailed to load music. setName=terrain_core:dirt	musicURI=
[0m[1;31mException: Resource with name MUSIC/DIRT and type SOUND wasn't found.
[0m[1;33mWarning: Cannot open Music/Dirt.mp3: 
[0m[1;31mUnable to play music (music parameter was NULL)
[0m[1;31mFailed to load music. setName=terrain_core:dirt	musicURI=
[0m[1;31mException: Resource with name MUSIC/DIRT and type SOUND wasn't found.
[0m[1;33mWarning: Cannot open Music/Dirt.mp3: music parameter was NULL
[0m[1;31mUnable to play music (music parameter was NULL)
[0m[1;31mFailed to load music. setName=	musicURI=
[0m[1;31mException: Resource with name MUSIC/TOWERTOWN and type SOUND wasn't found.
[0m[1;33mWarning: Cannot open music/TowerTown: music parameter was NULL
[0m[1;31mUnable to play music (music parameter was NULL)
[0m[1;31mThis line is called
[0m[swscaler @ 0x148460000] No accelerated colorspace conversion found from yuv420p to bgra.
[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mThis line is called
[0m[1;31mFailed to load music.```

Reply:
This is not a loop.
These are totalRedraw() calls that are used sometimes too often

Reply:
Sure, something like this would be more clear:
```c++
Rect targetRect( ... );
ColorRGBA colorToFill(...);
to.drawColor(targetRect, colorToFill);
```
But performance-wise there is nothing to be done here.

Reply:
Thanks together :)